engine/src/main/java/org/terasology/persistence/typeHandling/extensionTypes/ModifiableValueTypeHandler.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.persistence.typeHandling.extensionTypes;
+
+import gnu.trove.list.TFloatList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.persistence.typeHandling.PersistedData;
+import org.terasology.persistence.typeHandling.PersistedDataArray;
+import org.terasology.persistence.typeHandling.PersistedDataSerializer;
+import org.terasology.persistence.typeHandling.RegisterTypeHandler;
+import org.terasology.utilities.modifiable.ModifiableValue;
+
+import java.util.Optional;
+
+@RegisterTypeHandler
+public class ModifiableValueTypeHandler extends org.terasology.persistence.typeHandling.TypeHandler<ModifiableValue> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ModifiableValueTypeHandler.class);
+
+    @Override
+    public PersistedData serializeNonNull(ModifiableValue value, PersistedDataSerializer serializer) {
+        return serializer.serialize(value.getBaseValue(), value.getPreModifiers(), value.getMultipliers(),
+                value.getPostModifiers());
+    }
+
+    @Override
+    public Optional<ModifiableValue> deserialize(PersistedData data) {
+        if (data.isArray()) {
+            PersistedDataArray vals = data.getAsArray();
+            if (vals.isNumberArray()) {
+                TFloatList floatList = vals.getAsFloatArray();
+                ModifiableValue modifiableValue = new ModifiableValue(floatList.get(0));
+                if (floatList.size() > 1) {
+                    modifiableValue.setPreModifiers(floatList.get(1));
+                    modifiableValue.setMultipliers(floatList.get(2));
+                    modifiableValue.setPostModifiers(floatList.get(3));
+                }
+                return Optional.of(modifiableValue);
+            }
+        }
```

Comment:
I'm wondering whether it would make sense to check an `else if` here for a single number. That would allow to easily specify the base value in a prefab without worrying about all the modifiers. So you could either do 

```jsonc
"myValue": 7,					// base value only
// ... or ...
"otherValaue": [3, 0, 0.9, 2]	// base and modifiers/multiplier
```

Reply:
This is handled by default as the constructor takes the base value and assigns default modifiers. The single value specified in the prefabs is deserialized into the constructor for ModifiableValue which takes only the base value. Other modifiers can be set optionally